dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ErrorDecoder.java

```diff
@@ -96,6 +111,7 @@ RuntimeException decode(Response response) {
         } catch (NullPointerException | IOException e) {
```

Comment:
What about on 308/429 etc.? Those don't send those headers?

Reply:
We're adding the diagnostic info above in `decode` which delegates to this method, `decodeInternal`. Diagnostic data is applied to all exceptions returned by this method.